R/human_infection.R

```diff
@@ -289,27 +295,54 @@ calculate_treated <- function(
       replace = TRUE
     )
   ])
-
-  successful <- bernoulli_multi_p(parameters$drug_efficacy[drugs])
-  treated_index <- bitset_at(seek_treatment, successful)
-
-  # Update those who have been treated
-  if (treated_index$size() > 0) {
-    variables$state$queue_update('Tr', treated_index)
+  
+  if(!is.null(parameters$antimalarial_resistance)) {
+    antimalarial_resistance_drug_index <- as.numeric(parameters$antimalarial_resistance_drug)[drugs]
+    artemisinin_resistance_proportion <- vector()
+    early_treatment_failure_probability <- vector()
+    drug_indices <- list(); drug_index <- vector()
+    for(i in seq_along(parameters$antimalarial_resistance_drug)) {
+      drug_indices[[i]] <- which(drugs == i)
+      drug_index[i] <- which(parameters$antimalarial_resistance_drug == i)
+      artemisinin_resistance_proportion[drug_indices[[i]]] <- parameters$prop_artemisinin_resistant[[drug_index[i]]][match_timestep(ts = parameters$antimalarial_resistance_timesteps[[drug_index[i]]],
+                                                                                                                                    t = timestep)]
+      early_treatment_failure_probability[drug_indices[[i]]] <- parameters$early_treatment_failure_prob[[drug_index[i]]][match_timestep(ts = parameters$antimalarial_resistance_timesteps[[drug_index[i]]],
+                                                                                                                                        t = timestep)]
+    }
+    susceptible_to_treatment_index <- bernoulli_multi_p(p = 1 - (artemisinin_resistance_proportion * early_treatment_failure_probability))
+    drugs <- drugs[susceptible_to_treatment_index]
+    susceptible_to_treatment <- bitset_at(seek_treatment, susceptible_to_treatment_index)
+    n_ETF <- n_treat - susceptible_to_treatment$size()
```

Comment:
ETF will be unknown to most people. I'd probably call this `n_early_treatment_failure` to be very clear

Reply:
Agreed, n_ETF changed to n_early_treatment_failure both as an object in calculate_treated() and in the run_simulation() outputs